kodilla-testing/src/main/java/com/kodilla/testing/library/BookLibrary.java

```diff
@@ -0,0 +1,34 @@
+package com.kodilla.testing.library;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class BookLibrary {
+    LibraryDatabase libraryDatabase;
+
+    public BookLibrary(LibraryDatabase libraryDatabase) {
+        this.libraryDatabase = libraryDatabase;
+    }
+
+    public List<Book> listBooksWithCondition(String titleFragment) {
+        List<Book> bookList = new ArrayList<Book>();
+        if (titleFragment.length() < 3) return bookList;
+        List<Book> resultList = libraryDatabase
+                .listBooksWithCondition(titleFragment);
+        if (resultList.size() > 20) return bookList;
+        bookList = resultList;
+        return bookList;
+    }
+
+    // list books borrowed by libraryUser
+    public List<Book> listBooksInHandsOf(LibraryUser libraryUser) {
+        List<Book> listBooksInHandsOf = new ArrayList<Book>();
+        List<Book> resultList = libraryDatabase.listBooksInHandsOf(libraryUser);
+        if(resultList.size()==0) return listBooksInHandsOf;
+        if(resultList.size()>=1) return resultList;
+        listBooksInHandsOf = resultList;
```

Comment:
is it possible that both previous conditions won't be true and computer ends up here?
Think how to simplify this method